client/executor/wasmtime/src/instance_wrapper.rs

```diff
@@ -415,6 +415,43 @@ impl InstanceWrapper {
 			slice::from_raw_parts_mut(ptr, len)
 		}
 	}
+
+	/// Returns the pointer to the first byte of the linear memory for this instance.
+	pub fn base_ptr(&self) -> *const u8 {
+		self.memory.data_ptr()
+	}
+
+	/// Removes physical backing from the allocated linear memory. This leads to returning the memory
+	/// back to the system. While the memory is zeroed this is considered as a side-effect and is not
+	/// relied upon. Thus this function acts as a hint.
+	pub fn decommit(&self) {
+		if self.memory.data_size() == 0 {
+			return;
+		}
+
+		cfg_if::cfg_if! {
```

Comment:
We don't need `cfg_if` here?

Reply:
You mean that `madvise` is POSIX-compatible? Sure, but it's complicated. In Linux this is not merely an advice it has a very certain and widely relied semantics of eagerly freeing and zeroing the pages. macOS is, on the other hand, is very lazy about those. madvise(WONTNEED) won't have the same effect there.

There are different options there to make it work on macOS, but those are out of scope for this PR. When the time comes though it will be clear where to integrate them : )

Reply:
I mean we don't need the `cfg_if`.

Could be just `cfg(unix)`

Reply:
(Assuming you meant `cfg(target_os = "linux")`)

Yeah, true, not necessary. I figured I use it because:

- it's not a new dependency. We already use it and it is a very common dep that it occurs in our dep tree many times
- as I mentioned it's likely we want other branches for macOS and/or `cfg(unix)`

It's not something I am married to, so can change to a plain `cfg` as well. Just let me know

Reply:
Yeah fine